ament_tools/build_types/ament_python.py

```diff
@@ -369,14 +369,18 @@ def _undo_install(self, context):
         if os.path.exists(install_log):
             with open(install_log, 'r') as h:
                 lines = [l.rstrip() for l in h.readlines()]
-            directories = []
+            directories = set()
+            install_space = context.install_space + os.sep
             for line in lines:
                 if os.path.exists(line) and \
-                        line.startswith(context.install_space):
+                        line.startswith(install_space):
```

Comment:
Have you thought about what will happen when `install_space` has quotes, spaces, and slashes? (It might work fine still - I am investigating it myself, but you may have thought about it already)